### PR TITLE
PWN-8753 - Swap. Fix InvalidTimestamp from Whirlpools

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8753

## Description of Work

- see the comments

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmSendErrorConverter.kt [hunk 1]
```diff
@@ -219,8 +219,9 @@ class AlarmSendErrorConverter(
         userPublicKey: Base58String,
         error: Throwable
     ): AlarmErrorsRequest {
-        val errorDescription = when (error) {
+        val errorDescription: String = when (error) {
             is UsernameServiceError -> "${error.javaClass.simpleName}: ${error.errorCode}, ${error.message}"
+            is ServerException -> error.jsonErrorBody?.toString() ?: error.message ?: error.toString()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmSendErrorConverter.kt:224]: remove "ekranirovanie" when the error comes to alarm

File: app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt [hunk 2]
```diff
@@ -1,115 +1,51 @@
...
-        }
-    }
+            is SwapState.LoadingRoutes -> tokenA to tokenB
+            is SwapState.LoadingTransaction -> tokenA to tokenB
+            is SwapState.SwapLoaded -> tokenA to tokenB
+            is SwapState.TokenAZero -> tokenA to tokenB
+            is SwapState.TokenANotZero -> tokenA to tokenB
+            is SwapState.RoutesLoaded -> tokenA to tokenB
 
-    fun getTokenAAmount(state: SwapState): BigDecimal? {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt:91]: extracted as extensions, it's not interactor responsibility i think

File: app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapState.kt [hunk 3]
```diff
@@ -57,7 +57,7 @@ sealed interface SwapState {
         val lamportsTokenA: BigInteger,
         val lamportsTokenB: BigInteger,
         val routes: List<JupiterSwapRoute>,
-        val activeRoute: Int,
+        val activeRouteIndex: Int,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapState.kt:60]: finally good names!

File: app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapSendTransactionDelegate.kt [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    private val tokenKeyProvider: TokenKeyProvider,
+    private val relaySdkFacade: RelaySdkFacade
+) {
+    private var retryCount: Int = 0
+
+    suspend fun sendSwapTransaction(
+        swapRoute: JupiterSwapRoute,
+        jupiterTransaction: Base64String
+    ): JupiterSwapTokensResult = try {
+        retryCount++
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapSendTransactionDelegate.kt:33]: I can't find where the `retryCount` is set back to `0`, or is it unnecessary?

**Review by eduardmaximovich — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**